subsys/net/lib/fota_download/src/util/fota_download_util.c

```diff
@@ -0,0 +1,407 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* Ensure 'strnlen' is available even with -std=c99. */
+#if !defined(_POSIX_C_SOURCE)
+#define _POSIX_C_SOURCE 200809L
+#endif
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <string.h>
+#include <nrfx.h>
+
+#include "fota_download_util.h"
+
+#if defined(CONFIG_DFU_TARGET_MODEM_DELTA)
+#include "fota_download_delta_modem.h"
+#endif
+
+#if defined(CONFIG_DFU_TARGET_FULL_MODEM)
+#include "fota_download_full_modem.h"
+#endif
+
+#if defined(CONFIG_DFU_TARGET_MCUBOOT)
+#include "fota_download_mcuboot.h"
+#endif
+
+#if defined(CONFIG_DFU_TARGET_SMP)
+#include "fota_download_smp.h"
+#endif
+
+#define MAX_RESOURCE_LOCATOR_LEN 500
```

Comment:
Maybe a Kconfig too?

Reply:
This is old part of code. But if it make sense to but kConfig

Reply:
Yes it make sense to use kConfig it make things safer